### PR TITLE
Fix non-static auto issue

### PR DESCRIPTION
**Pull request by TheCynosure**
Because GCC decided to now enforce this standard this breaks the compilation for the new versions of gcc and g++. This should fix the issue.

### CODE REVIEW

**Comment by kylevedder** [comment]
Can you squash these into a single commit? You ought to be able to rebase onto `master` and stick this as a single diff.

File: include/config_reader/macros.h [hunk 1]
```diff
@@ -37,60 +37,62 @@ namespace config_reader {
...
 
 #define CONFIG_INT(name, key)                                                  \
-  const auto& MAKE_NAME(name) =                                                \
+  const int& MAKE_NAME(name) =                                                 \
       ::config_reader::InitVar<int, ::config_reader::config_types::ConfigInt>( \
           key)
-#define CONFIG_UINT(name, key)                            \
-  const auto& MAKE_NAME(name) = ::config_reader::InitVar< \
+#define CONFIG_UINT(name, key)                                      \
+  const(unsigned int) & MAKE_NAME(name) = ::config_reader::InitVar< \
```
Comment (kylevedder) [inline include/config_reader/macros.h]: Why is `(unsigned int)` parenthesized?

Comment (TheCynosure) [reply include/config_reader/macros.h]: Because it is a type using two seperate words. I can remove the paranethsis but in my mind it made more sense.

**Comment by TheCynosure** [comment]
I stink at Git, but my only option was to squash into your Eigen commit (probably could have done it differently but here we are). So it should all be squashed down, I didn't want to resolve the conflicts either (its the section without auto) because it said it would auto-merge it which I want you to confirm its ok.

**Comment by kylevedder** [comment]
You reverted one commit too far, and now your commit is essentially being smooshed into the same commit as the one that added the `Eigen::Vector2f` and `Eigen::Vector3f` support. If memory serves the changes you made were pretty straightforward; my suggestion is reset your git repo to what's what's on `master` right now, branch that, make your changes, and then commit that as a single commit on top of what's on master.

Regarding the `(unsigned int)`: I've never seen that done anywhere before and I can't think of why it would be safer, so I'd prefer you dropped the parenthesis unless there's a reason you think it's safer.

**Comment by TheCynosure** [comment]
Ok I am just going to close this and redo the changes to be correct and then will open a new PR.

**Pull request closed without merging**